### DISCUSSION TITLE
Keeping track of where this function was called from

### DISCUSSION THREAD

**Discussion by valkyrienyanko** [Valks-Games/sankari#191] [General]
I've added this `string raycastGroup` to help with debugging and I'm thinking of using this convention in other functions but it's kind of tedious to setup. Is there a easier way of keeping track of something like this?

![image](https://user-images.githubusercontent.com/6277739/199087564-d3b61931-f38b-4d1f-ad83-b3795ffd1c0b.png)

I would have to add a string param here too if I wanted to do the same thing but again it's seems like there is a better way of doing this

![image](https://user-images.githubusercontent.com/6277739/199087619-71f433b0-7d70-43e1-b558-e1226a9fa349.png)

**Comment by policiu** [comment]
Probably not too worth it. If we log it with a stack trace, it should be good enough. If we do it, I'd perfer:

```
protected bool IsNearWallLeft() => AreRaycastsColliding(RaycastsWallLeft, nameof(RaycastsWallLeft));
```

If you really want to, the best idea I have is to create a Wrapper around the list with a name property. I wouldn't recommend it.

```csharp
	private class NamedList<T> : List<T>
	{
		public string Name { get; set; } = "Hello!";
	}

	private void Tester()
	{
		NamedList<string> fruits = new()
		{
			Name = "Basket of Fruit"
		};
		CheckFruit(fruits);
	}

	private void CheckFruit(NamedList<string> fruits)
	{
		Console.WriteLine(fruits.Name);

		fruits.ForEach(fruit => fruit += "!");
	}
```
